state-chain/chains/src/sol/api.rs

```diff
@@ -222,14 +237,32 @@ impl<Env: 'static> ExecutexSwapAndCall<Solana> for SolanaApi<Env> {
 	}
 }
 
-impl<Env: 'static> AllBatch<Solana> for SolanaApi<Env> {
+impl<Env: SolanaEnvironment> AllBatch<Solana> for SolanaApi<Env> {
 	fn new_unsigned(
-		_fetch_params: vec::Vec<FetchAssetParams<Solana>>,
-		_transfer_params: vec::Vec<TransferAssetParams<Solana>>,
+		fetch_params: vec::Vec<FetchAssetParams<Solana>>,
+		transfer_params: vec::Vec<TransferAssetParams<Solana>>,
 	) -> Result<Self, AllBatchError> {
-		// Create a BatchFetch for all deposit_channels
-		// Figure out how to derive deposit_channels from ChannelId (or pass DepositChannel in)
-		// Create a Transfer for each Transfer
+		let vault_program = Env::lookup_account(SolanaEnvAccountLookupKey::VaultProgram)
+			.map_err(|e| AllBatchError::DispatchError(e.into()))?;
+		let deposit_channels = fetch_params
+			.into_iter()
+			.map(|fetch_param| {
+				address_derivation::derive_deposit_channel::<Env>(
```

Comment:
We shouldn't have to re-derive the deposit channel here. 

I think maybe we need to change the definition of the `DepositFetchId` for Solana. 

At the moment it's just `ChannelId` - we can make it `(ChannelId, Bump)`, and then we have all the information we need to build the fetch tx.

Reply:
We still need the derived account ID for fetching as well. What we need is (AccountID, Bump), I feel like it is easier to derive it from a channelID than storing the accountID + bump and passing them in

Reply:
I would prefer if we follow existing patterns: we already have derived all the information and it's already stored in the ingress-egress pallet where we trigger the fetch. For all other chains we define the DepositFetchId such that it contains all the information required for a fetch, and we extract this from the DepositChannel (hence this constraint on the DepositFetchId: 
```
		+ for<'a> From<&'a DepositChannel<Self>>;
```

I think the correct approach here is to use the DepositFetchId for this, ie. create a SolanaDepositFetchId with the address, channel_id and bump.

Then the FetchAssetParams will contain the required information and we don't need this extra conversion into a DepositChannel.

Reply:
Alternatively we could refactor fetching such that it always passes a list of deposit channels, and remove the whole layer of abstraction around FetchId etc. I think this might make things clearer and easier to implement. But this would be a much bigger refactor. 

Either way, we should use an approach that is consistent across chains.

Reply:
Done. We only pass in FetchParam, instead of re-generating the deposit_channel